tests/lib/rules/yoda.js

```diff
@@ -992,6 +1017,110 @@ ruleTester.run("yoda", rule, {
                     type: "BinaryExpression"
                 }
             ]
+        },
+        {
+            code: "if (`green` < x.y && x.y < `blue`) {}",
+            output: "if (`green` < x.y && `blue` > x.y) {}",
+            options: ["always", { exceptRange: true }],
+            parserOptions: { ecmaVersion: 2015 },
+            errors: [
+                {
+                    messageId: "expected",
+                    data: { expectedSide: "left", operator: "<" },
+                    type: "BinaryExpression"
+                }
+            ]
+        },
+        {
+            code: "if (x < -1n || 1n <= x) {}",
+            output: "if (x < -1n || x >= 1n) {}",
+            options: ["never", { exceptRange: true }],
+            parserOptions: { ecmaVersion: 2020 },
+            errors: [
+                {
+                    messageId: "expected",
+                    data: { expectedSide: "right", operator: "<=" },
+                    type: "BinaryExpression"
+                }
+            ]
+        },
+        {
+            code: "if('a' <= x && x < 'b') {}",
+            output: "if('a' <= x && 'b' > x) {}",
+            options: ["always"],
+            errors: [
+                {
+                    messageId: "expected",
+                    data: { expectedSide: "left", operator: "<" },
+                    type: "BinaryExpression"
+                }
+            ]
+        },
+        {
+            code: "if (x < -1n || 1n <= x) {}",
```

Comment:
false negative . 

I will take a look 👍

Reply:
> `if (x < -1n || 1n <= x) {}`

this is not a `yoda` and `always` allows only yoda but with `exceptRange` options it should be valid (lint free)


_looking into it 👍 _

Reply:
fixed